fvm/state/accounts.go

```diff
@@ -407,42 +378,69 @@ func (a *Accounts) getContractNames(address flow.Address) (contractNames, error)
 	return identifiers, nil
 }
 
-func (a *Accounts) GetContract(name string, address flow.Address) ([]byte, error) {
+func (a *Accounts) GetContract(contractName string, address flow.Address) ([]byte, error) {
 	contractNames, err := a.getContractNames(address)
 	if err != nil {
 		return nil, err
 	}
-	if !contractNames.Has(name) {
+	if !contractNames.Has(contractName) {
 		return nil, nil
 	}
-	return a.getCode(name, address)
+	return a.getContract(contractName, address)
 }
 
-func (a *Accounts) SetContract(name string, address flow.Address, code []byte) error {
+func (a *Accounts) SetContract(contractName string, address flow.Address, contract []byte) error {
 	contractNames, err := a.getContractNames(address)
 	if err != nil {
 		return err
 	}
-	err = a.setCode(name, address, code)
+	err = a.setContract(contractName, address, contract)
 	if err != nil {
 		return err
 	}
-	contractNames.add(name)
+	contractNames.add(contractName)
 	return a.setContractNames(contractNames, address)
 }
 
-func (a *Accounts) DeleteContract(name string, address flow.Address) error {
+func (a *Accounts) DeleteContract(contractName string, address flow.Address) error {
 	contractNames, err := a.getContractNames(address)
 	if err != nil {
 		return err
 	}
-	if !contractNames.Has(name) {
+	if !contractNames.Has(contractName) {
 		return nil
 	}
-	err = a.setCode(name, address, nil)
+	err = a.setContract(contractName, address, nil)
 	if err != nil {
 		return err
 	}
-	contractNames.remove(name)
+	contractNames.remove(contractName)
 	return a.setContractNames(contractNames, address)
 }
+
+// contractNames container for a list of contract names. Should always be sorted.
+// To ensure this, don't sort while reading it from storage, but sort it while adding/removing elements
+type contractNames []string
```

Comment:
Maybe I miss something, but do we use this list for anything else than checking membership?

Reply:
same reply as here: https://github.com/onflow/flow-go/pull/7#discussion_r508674384